storage_service/locations/models/s3.py

```diff
@@ -76,7 +76,13 @@ def resource(self):
         return self._resource
 
     def _ensure_bucket_exists(self):
-        self.client.create_bucket(Bucket=self._bucket_name())
+        try:
+            self.client.head_bucket(Bucket=self._bucket_name())
+        except self.client.exceptions.ClientError:
+            self.client.create_bucket(
```

Comment:
I think this is still missing the check that the error was a 404:

```
            # If a client error is thrown, then check that it was a 404 error.
            # If it was a 404 error, then the bucket does not exist.
            error_code = e.response["Error"]["Code"]
            if error_code == "404":
                 ...
```

Reply:
:+1: